components/org.wso2.micro.integrator.core/src/main/java/org/wso2/micro/integrator/core/internal/Activator.java

```diff
@@ -162,4 +170,24 @@ private void logServerInfo() {
 
     }
 
+    private static void handleConfiguration() {
+
+        String resourcesDir = Paths.get(System.getProperty(LauncherConstants.CARBON_HOME),
+                                        "repository", "resources", "conf").toString();
+
+        String configFilePath = System.getProperty(DEPLOYMENT_CONFIG_FILE_PATH);
+        if (StringUtils.isEmpty(configFilePath)) {
+            configFilePath = Paths.get(System.getProperty(LauncherConstants.CARBON_CONFIG_DIR_PATH),
+                                       ConfigParser.UX_FILE_PATH).toString();
+        }
+
+        String outputDir = System.getProperty(LauncherConstants.CARBON_HOME);
+        try {
+            ConfigParser.parse(configFilePath, resourcesDir, outputDir);
+        } catch (ConfigParserException e) {
+            log.info("Error while performing configuration changes", e);
+            System.exit(1);
```

Comment:
Should we exit here? Or throw the exception?

Reply:
This was early discussed and decided to stop execution and show severe error